File: +VehicleDynamicsLateral/@TireLinear/TireLinear.m
classdef TireLinear < VehicleDynamicsLateral.Tire
    % TireLinear Linear tire model
    %
    % It inherits methods from Tire.

    methods
        % Constructor
        function self = TireLinear()
            self.k = 40000;
        end

        function Fy = Characteristic(self, alpha, varargin)
            Fy = -self.k * alpha;
        end
    end

    properties
        k % Cornering stiffness [N/rad]
    end

end

File: pyvd/TireLinear.py
import Tire


class TireLinear(Tire.Tire):
    '''TireLinear Linear tire model

    :var k: Cornering stiffness [N/rad]
    :var q: Cornering stiffness 2 [N/rad]

    '''

    def __init__(self):
        super(TireLinear, self).__init__()
     # Tire.Tire.__init__(self)

    def PlotTire(self):
        ''' Returns the handle of the curve'''
        # alpha = (0:0.1:15)*pi/180
        # Fy = - self.Characteristic(alpha)
        # p = plot(alpha*180/pi,Fy)
        # grid on; box on;
        # xlabel('Slip angle [deg]')
        # ylabel('Lateral force [N]')

    def Characteristic(self, alpha, varargin):
        # ''' Returns the handle of the curve'''
        # Fy = - self.k * alpha
        pass
